Sources/Fuzzilli/FuzzIL/AbstractInterpreter.swift

```diff
@@ -11,17 +11,116 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-
-/// Analyzes the types of variables.
-public struct AbstractInterpreter {
-    // States are kept in a stack to support conditional execution.
-    private var ifStack: [VariableMap<Type>] = []
-    private var stack = [VariableMap<Type>()]
+private struct InterpreterState {
+    // Save variable changes, supports
+    // 1. Nested states (function states, loop states)
+    // 2. Sibling states (if/else branches)
+    private var stateChangesStack = [[VariableMap<Type>()], [VariableMap<Type>()]]
```

Comment:
Can you make a short comment here why the stack starts with two states? I assume it's because you always need the "parent" state and the active state?